src/core/plugins/systemimage/api.js

```diff
@@ -43,9 +43,17 @@ const gpgFiles = [
 const getImages = (channel, device, wipe, enable = [], disable = []) =>
   api
     .get(`${channel}/${device}/index.json`)
-    .then(({ data }) => data.images.filter(({ type }) => type === "full"))
-    .then(images => ({
-      files: images[images.length - 1].files
+    .catch(error => {
+      if (error?.response?.status === 404) throw new Error("404");
+      else throw new Error("no network");
```

Comment:
Should probably just use https://www.electronjs.org/docs/latest/tutorial/online-offline-events ?
There are a lot of other errors that can happen here that are not network related - might be better to tell the user to try again ;p